templates/zerver/desktop_redirect.html

```diff
@@ -1,18 +1,17 @@
 {% extends "zerver/portico.html" %}
-
-{% block customhead %}
-<meta http-equiv="Refresh" content="2; {{ desktop_url }}">
-{% endblock %}
+{% set entrypoint = "desktop-redirect" %}
 
 {% block content %}
 <div class="flex">
     <div class="desktop-redirect-box">
         <img class="avatar desktop-redirect-image" src="{{ realm_icon_url }}" alt=""/><br>
-        Redirecting to your Zulip app...
-        <div class="desktop-redirect-links">
-            <a id="desktop-redirect-link" href="{{ desktop_url }}">If you weren't redirected, click here.</a><br>
-            <a href="{{ browser_url }}">Or, continue in your browser.</a>
-        </div>
+        <p>{% trans %}Copy this login token and return to your Zulip app to finish logging in:{% endtrans %}</p>
+        <p>
+            <input id="desktop-data" value="{{ desktop_data }}" readonly />
+            <button id="copy" tabindex="0" class="btn btn-primary" data-clipboard-target="#desktop-data">{% trans %}Copy{% endtrans %}</button>
+        </p>
+        <p>{% trans %}You may then close this window.{% endtrans %}</p>
```

Comment:
Do we need to clarify one doesn't need to paste, just copy?

Reply:
Only the app page knows whether the user will need to paste. I don’t think users will complain that we made the flow easier than they expected.

In fact, I’m considering showing a placebo paste box even in the automatic case, just to gently confirm to the user that copying the token is correct.